### ISSUE TITLE
Multiple .db files in /tmp

### ISSUE THREAD

**Issue by Jean-PhilippeD** [prometheus/client_python#971]
Hello,

I'm using prometheus client inside a docker and I got a strange issue.
When running outside docker with `PROMETHEUS_MULTIPROC_DIR ` set to /tmp/prometheus.

It's working fine.

Running inside docker (even inside a K8S cluster), I see many many db files under /tmp and nothing in /tmp/prometheus.

```
-rw-r--r--    1 app   app   65536 Oct 25 10:10 histogram_107279.db
-rw-r--r--    1 app   app   65536 Oct 25 10:10 counter_107279.db
-rw-r--r--    1 app   app   65536 Oct 25 10:11 histogram_105945.db
-rw-r--r--    1 app   app   65536 Oct 25 10:11 counter_105945.db
-rw-r--r--    1 app   app   65536 Oct 25 10:12 histogram_105188.db
-rw-r--r--    1 app   app   65536 Oct 25 10:12 counter_105188.db
-rw-r--r--    1 app   app   65536 Oct 25 10:12 histogram_104407.db
-rw-r--r--    1 app   app   65536 Oct 25 10:12 counter_104407.db
-rw-r--r--    1 app   app   65536 Oct 25 10:13 histogram_101715.db
-rw-r--r--    1 app   app   65536 Oct 25 10:13 counter_101715.db
-rw-r--r--    1 app   app   65536 Oct 25 10:14 histogram_112498.db
-rw-r--r--    1 app   app   65536 Oct 25 10:14 counter_112498.db
-rw-r--r--    1 app   app   65536 Oct 25 10:14 histogram_113638.db
-rw-r--r--    1 app   app   65536 Oct 25 10:14 counter_113638.db
-rw-r--r--    1 app   app   65536 Oct 25 10:14 histogram_103384.db
-rw-r--r--    1 app   app   65536 Oct 25 10:14 counter_103384.db
```

And those db still increments over the time minutes by minutes.. 
I have no ideas what's happening.

**Comment by Jean-PhilippeD** [comment]
maybe it's a missunderstanding for myself, is it the normal behaviour ??

**Comment by csmarchbanks** [comment]
This is normal, when running in multiprocess mode each process creates its own .db file for each metric type. When the metrics are exposed the process handling the request reads all of the .db files and aggregates the results together.

I am going to close this issue, but if you are running into issues let me know!

**Comment by Jean-PhilippeD** [comment]
So it creates a new .db file for the same metric for each new value of the metric ? So it grows over and over .. ?

But in my case, I try to do a gauge.set(0) and I never see 0, is-it because of the other files remaining ???

**Comment by csmarchbanks** [comment]
Not for each value, but for each process id. So the same process will keep updating its .db files with new values.

Are you using a `multiprocess_mode` for your gauge?

**Comment by Jean-PhilippeD** [comment]
Yes I use it with Celery. I disabled auto scaling.
But I just found that you've added a new policy for Gauge (mostrecent), I think that's what I need to monitor my metric.

I have a Gauge counting the task running.
I increment when a task start, I decrement when a task stopped.
But everything is asynchronous, so when a worker restart or is upgraded, the counter may be false.

So with `mostrecent`, I will not use `inc` or `dec` but only set with the true value from my tasks running in database, and prometheus will keep the fresh values.  I could not use this way without `mostrecent` policy.
